cel/pkg/reconciler/cel/celrun.go

```diff
@@ -0,0 +1,121 @@
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cel
+
+import (
+	"context"
+	"fmt"
+	"github.com/google/cel-go/cel"
+	"github.com/google/cel-go/common/types"
+	"github.com/tektoncd/pipeline/pkg/client/injection/reconciler/pipeline/v1alpha1/run"
+	"knative.dev/pkg/logging"
+
+	v1alpha1 "github.com/tektoncd/pipeline/pkg/apis/pipeline/v1alpha1"
+	v1 "k8s.io/api/core/v1"
+	reconciler "knative.dev/pkg/reconciler"
+)
+
+// newReconciledNormal makes a new reconciler event with event type Normal, and reason RunReconciled.
+func newReconciledNormal(namespace, name string) reconciler.Event {
+	return reconciler.NewEvent(v1.EventTypeNormal, "RunReconciled", "Run reconciled: \"%s/%s\"", namespace, name)
+}
+
+// Reconciler implements controller.Reconciler for Run resources.
+type Reconciler struct {
+}
+
+// Check that our Reconciler implements Interface
+var _ run.Interface = (*Reconciler)(nil)
+
+// ReconcileKind implements Interface.ReconcileKind.
+func (r *Reconciler) ReconcileKind(ctx context.Context, o *v1alpha1.Run) reconciler.Event {
+	logger := logging.FromContext(ctx)
+	logger.Infof("Reconciling Run %s/%s", o.Namespace, o.Name)
+
+	// If the Run has not started, initialize the Condition and set the start time.
+	if !o.HasStarted() {
+		logger.Infof("Starting new Run %s/%s", o.Namespace, o.Name)
+		o.Status.InitializeConditions()
+		// In case node time was not synchronized, when controller has been scheduled to other nodes.
+		if o.Status.StartTime.Sub(o.CreationTimestamp.Time) < 0 {
+			logger.Warnf("Run %s/%s createTimestamp %s is after the Run started %s", o.Namespace, o.Name, o.CreationTimestamp, o.Status.StartTime)
+			o.Status.StartTime = &o.CreationTimestamp
+		}
+	}
+
+	if o.IsDone() {
+		logger.Infof("Run %s/%s is done", o.Namespace, o.Name)
+		return nil
+	}
+
+	// Create a program environment configured with the standard library of CEL functions and macros
+	env, err := cel.NewEnv(cel.Declarations())
+	if err != nil {
+		logger.Errorf("cel.NewEnv: %v", err)
+		return err
+	}
+
+	// Iterate through the CEL expressions
+	for _, param := range o.Spec.Params {
+		if param.Value.StringVal == "" {
+			logger.Errorf("Invalid CEL expression parameter %s in Run %s/%s", param.Name, o.Namespace, o.Name)
+			o.Status.MarkRunFailed("InvalidExpression",
+				"CEL expression parameter %s was invalid in Run %s/%s", param.Name, o.Namespace, o.Name)
+			return nil
+		}
+
+		// Combine the Parse and Check phases CEL program compilation to produce an Ast and associated issues
+		ast, iss := env.Compile(param.Value.StringVal)
+		if iss.Err() != nil {
+			logger.Errorf("CEL expression %s could not be parsed when reconciling %s/%s", param.Name, o.Namespace, o.Name)
+			o.Status.MarkRunFailed("SyntaxError",
+				"CEL expression %s could not be parsed in Run %s/%s", param.Name, o.Namespace, o.Name, err)
```

Comment:
ditto on including name+namespace, but make sure to include the error message (and in the `logger.Errorf` on the previous line).

And same below, you can remove the references to `o.Name` and `o.Namespace`.